pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java

```diff
@@ -98,6 +100,21 @@ static DocIdTranslatorMode of(String mode) {
     }
   }
 
+  public TextIndexConfig(Boolean disabled, FSTType fstType, Object rawValueForTextIndex, boolean enableQueryCache,
```

Comment:
(minor) I don't think we need to keep this default constructor. There are not that many usages, and we should probably change them to use builder pattern

Reply:
This is required now by the recent spi backward compatibility check.
It doesn't have a good way to exclude those jsonCreator constructors.

Reply:
Hmm, in that case I feel it is just too strict :(
I feel we can ignore the test failure since we know exactly why it fails